Programmatic Settings UI

Tom-Todd opened the pull request:
Changed the Settings UI to be generated Programmatically.

elken committed:
Update CONTRIBUTING.md

Hal9007 committed:
Merge remote-tracking branch 'origin/dev'

Tom-Todd committed:
Merge branch 'master' of github.com:tom-todd/client into NewsFeed

Tom-Todd committed:
Merge remote-tracking branch 'upstream/dev' into NewsFeed

Tom-Todd committed:
Changed Settings Widget UI to be Programmatically defined as opposed to XML.

Tom-Todd committed:
Settings Widget:

-Reformatted settings files
-Changed slot names to be more descriptive

Tom-Todd committed:
Settings Widget:

-Added Labels back to group boxes
-Cleaned up styles.

Tom-Todd committed:
Settings Widget:

-Added Spacing in the horizontal layout.

Tom-Todd committed:
Settings Widget:

-Removed Redundant settings.ui

lolzballs commented:
When creating new objects with `new`, set the parent to the parent widget, so that Qt can delete the child widget with the parent widget is deleted.

Tom-Todd committed:
Settings Widget:

-Added widget as parent of objects.

Tom-Todd commented:
Changed objects to have parent.

Tom-Todd committed:
Settings Widget:

-Fixed inheritance issues with layouts
-Added buttons to set text colours (not implemented).

Tom-Todd committed:
Settings Widget:

-Fixed formatting issues.

Tom-Todd committed:
Settings Widget:

-Removed commented ui declararion in Settings.cpp
